src/snap.py

```diff
@@ -45,7 +45,7 @@ def install(self) -> bool:
         """
         try:
             apt.update()
-            apt.add_package(["snapd", "openjdk-17-jre-headless"])
+            apt.add_package(["snapd"])
```

Comment:
**question** is it common for all the charms to have this? Could we skip the snapd package and rely on the one provided by the OS already?

Reply:
Common for all. Why? I think it's just to ensure it, but you're right, it's not strictly necessary.

Reply:
I mean, we could - in theory - have the same problem we had with `openjdk`, and not have stable deployments

Reply:
If we couldn't pull snapd, we couldn't pull the zk snap either, so it's likely a non-issue.

Reply:
But it is not that we could not pull, it was rather that what we pulled had a broken patch. In the charm, we pin the SNAP revision, and deployment should always be stable

Reply:
It does nothing if it's already installed.

```
root@ubuntu:~# apt install snapd
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
snapd is already the newest version (2.58+22.04.1).
snapd set to manually installed.
The following package was automatically installed and is no longer required:
  libfreetype6
Use 'apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

However leaving it in is explicitly 'ensuring' that it's always there. Although we only 'support' on Ubuntu, there's no harm to leaving it here IMO in case that 'implicit' requirement changes down the line.